Components/Motors/VESC.cs

```diff
@@ -9,31 +9,97 @@ namespace Scarlet.Components.Motors
 {
     public class VESC : IMotor
     {
-        private IFilter<float> Filter; // Filter for speed output
+        #region enums
+        private enum UARTPacketID : byte
+        {
+            // Full enum list here: https://github.com/vedderb/bldc_uart_comm_stm32f4_discovery/blob/master/datatypes.h
+            FW_VERSION = 0,
+            GET_VALUES = 4,
+            SET_DUTY = 5,
+            SET_CURRENT = 6,
+            SET_CURRENT_BRAKE = 7,
+            SET_RPM = 8,
+            SET_POS = 9,
+            SET_DETECT = 10,
+            REBOOT = 28,
+            ALIVE = 29,
+            FORWARD_CAN = 33,
+        }
+
+        private enum CANPacketID : byte
+        {
+            CAN_PACKET_SET_DUTY = 0,
+            CAN_PACKET_SET_CURRENT = 1,
+            CAN_PACKET_SET_CURRENT_BRAKE = 2,
+            CAN_PACKET_SET_RPM = 3,
+            CAN_PACKET_SET_POS = 4,
+            CAN_PACKET_FILL_RX_BUFFER = 5,
+            CAN_PACKET_FILL_RX_BUFFER_LONG = 6,
+            CAN_PACKET_PROCESS_RX_BUFFER = 7,
+            CAN_PACKET_PROCESS_SHORT_BUFFER = 8,
+            CAN_PACKET_STATUS = 9,
+        }
+        #endregion
+
+        private readonly sbyte MOTOR_MAX_RPM;
+        private readonly int ERPM_PER_RPM;
+
+        private IFilter<sbyte> RPMFilter; // Filter for speed output
         private readonly IUARTBus UARTBus;
+        private readonly ICANBus CANBus;
         private readonly int CANForwardID;
-        private readonly float MaxSpeed;
+        private readonly uint CANID;
+        private readonly sbyte MaxRPM;
+        private readonly bool IsCAN;
 
         private bool OngoingSpeedThread; // Whether or not a thread is running to set the speed
         private bool Stopped; // Whether or not the motor is stopped
         public float TargetSpeed { get; private set; } // Target speed (-1.0 to 1.0) of the motor
+        public sbyte TargetRPM { get; private set; } // Target RPM of the motor
 
         /// <summary> Initializes a VESC Motor controller </summary>
         /// <param name="UARTBus"> UART output to control the motor controller </param>
         /// <param name="MaxSpeed"> Limiting factor for speed (should never exceed + or - this val) </param>
         /// <param name="CANForwardID"> CAN ID of the motor controller (-1 to disable CAN forwarding) </param>
-        /// <param name="SpeedFilter"> Filter to use with MC. Good for ramp-up protection and other applications </param>
-        public VESC(IUARTBus UARTBus, float MaxSpeed, int CANForwardId = -1, IFilter<float> SpeedFilter = null)
+        /// <param name="RPMFilter"> Filter to use with MC. Good for ramp-up protection and other applications </param>
+        public VESC(IUARTBus UARTBus, float MaxSpeed, int CANForwardID = -1, IFilter<sbyte> RPMFilter = null, int ErpmPerRpm = 518, sbyte MotorMaxRpm = 60)
```

Comment:
`ErpmPerRpm` and `MotorMaxRpm` probably shouldn't have default values unless these are specific to the motor controller. Same for the other ctor.